tests/filecheck/backend/riscv/register_allocation_liveness_block_naive.mlir

```diff
@@ -1,41 +1,41 @@
 // RUN: xdsl-opt -p riscv-allocate-registers{allocation_strategy=LivenessBlockNaive} %s --print-op-generic | filecheck %s
 
 "builtin.module"() ({
-  %0 = "riscv.li"() {"immediate" = 6 : i32} : () -> !riscv.reg<>
+  %0 = "riscv.li"() {"immediate" = 6 : i32} : () -> !riscv.reg<x$>
```

Comment:
Why this change in the generic printing?

Reply:
I thought we had agreed not to change the attribute printing for this PR

Reply:
Unallocated register no longer print as empty string, explicit ```x$``` is used. Internally, is ```None```.

Reply:
Here's a message in an earlier thread: https://github.com/xdslproject/xdsl/pull/1205#discussion_r1254216165 I may have marked it as resolved too early, it's my bad. I think that we should try to minimise the changes here, and make them only for custom printing of operations in the riscv dialect. All other changes should be merged as separate small PRs before or after this one. In the case of printing of RegisterType, let's merge it after.

Reply:
I agree with Tobias that it makes sense to keep the long signatures as-is as a first step. They are currently unambiguous in whether they're allocated or not.